library/x509_crt.c

```diff
@@ -657,8 +659,11 @@ static int x509_get_subject_alt_name( unsigned char **p,
                     MBEDTLS_ERR_ASN1_UNEXPECTED_TAG );
         }
 
-        /* Skip everything but DNS name */
-        if( tag != ( MBEDTLS_ASN1_CONTEXT_SPECIFIC | 2 ) )
+        tag_mask = tag & ( MBEDTLS_ASN1_TAG_CLASS_MASK | MBEDTLS_ASN1_TAG_VALUE_MASK );
+
+        /* Skip everything but DNS name and Other Name */
+        if( tag_mask != ( MBEDTLS_ASN1_CONTEXT_SPECIFIC | 2 ) &&
+            tag_mask != ( MBEDTLS_ASN1_CONTEXT_SPECIFIC | 0 )   )
```

Comment:
Could this check and the mask before be simplified slightly?

1. In line 653 we are already checking that the tag is CONTEXT_SPECIFIC, so here we would only need to check that the value is either dNSName or otherName, so I think the bitwise or is probably redundant
2. I see from the standard that otherName has value 0 and dNSName is 2. Its likely that we will add parsing for other types in the future. Could we just put these values in macros so that its easier to read  the code and we know where they are coming from?

Reply:
I second that, also for usability: Now that we support multiple name types, the user must inspect the `tag` field in each entry of the returned ASN.1 sequence, and we should provide macro constants for that.

Reply:
ok